### PR TITLE
feat(tracing): added hook to integrate with pino logger

### PR DESCRIPTION
**Pull request by CptSchnitz**

### CODE REVIEW

File: src/tracing/loggerHook.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  }
+  const traceObj = { spanId: span.context().spanId, traceId: span.context().traceId };
+
+  if (typeof args[0] === 'object') {
+    const [obj, ...rest] = args;
+    method.apply<Logger, Args, void>(this, [{ ...obj, ...traceObj }, ...rest]);
+  } else {
+    const [msg, ...rest] = args;
+    method.apply<Logger, Args, void>(this, [traceObj, msg, ...rest]);
+  }
```
Comment (syncush) [inline src/tracing/loggerHook.ts:23]: build args once and then use `method.apply` once

File: tests/tracing/loggerHook.spec.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    await provider.shutdown();
+  });
+
+  it('should add traceId and spanId to the logging only obj', async () => {
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    const args = [{ test: faker.random.word() }, faker.random.word()] as [object, string];
+    const logFn = jest.fn();
+
+    const provider = new NodeTracerProvider();
+    provider.register({ contextManager: new AsyncHooksContextManager() });
```
Comment (syncush) [inline tests/tracing/loggerHook.spec.ts:43]: lets register once

**Review by syncush — CHANGES_REQUESTED** [review]

File: src/tracing/loggerHook.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { getSpan, context } from '@opentelemetry/api';
+import { LogFn, Logger } from 'pino';
+
+// ignored because its the same type needed for LogFn
+// eslint-disable-next-line @typescript-eslint/ban-types, @typescript-eslint/no-explicit-any
+export type Args = [obj: object, msg?: string, ...args: any[]] | [msg: string, ...args: any[]];
+
+export function logMethod(this: Logger, args: Args, method: LogFn, level: number): void {
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply src/tracing/loggerHook.ts]: done

File: src/tracing/loggerHook.ts (hunk 1 above)
Comment (CptSchnitz) [reply src/tracing/loggerHook.ts]: done

File: tests/tracing/loggerHook.spec.ts (hunk 2 above)
Comment (CptSchnitz) [reply tests/tracing/loggerHook.spec.ts]: done

**Review by syncush — APPROVED** [review]

**Pull request merged**